src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -0,0 +1,243 @@
+            var startCharacter = linePosition.Character;
+
+            // 1. Token line number delta, relative to the previous token
+            var deltaLine = lineNumber - lastLineNumber;
+            Contract.ThrowIfTrue(deltaLine < 0);
+
+            // 2. Token start character delta, relative to the previous token
+            // (Relative to 0 or the previous token’s start if they're on the same line)
+            var startCharacterDelta = startCharacter;
+            if (lastLineNumber == lineNumber)
+            {
+                startCharacterDelta = startCharacter - lastStartCharacter;
+            }
+
+            // 3. Token length
+            var tokenLength = textSpan.Length;
+
+            // Getting the classified spans that are modifiers.
+            // Since a TextSpan can have multiple ClassifiedSpans, all of the ClassifiedSpans except one (the token)
+            // should be a modifier.
+            var modifiers = textSpanToClassifiedSpans.Where(
+                s => ClassificationTypeNames.AdditiveTypeNames.Contains(s.ClassificationType));
+
+            // Filtering out the modifiers for now (will be added in step 5). We just want the primary token.
+            var tokenTypeClassifiedSpan = textSpanToClassifiedSpans.Except(modifiers).Single();
+
+            // 4. Token type - looked up in SemanticTokensLegend.tokenTypes (language server defined mapping
+            // from integer to LSP token types).
+            var tokenTypeIndex = GetTokenTypeIndex(tokenTypeClassifiedSpan, tokenTypesToIndex);
+
+            // 5. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
+            var modifierBits = GetTokenTypeModifierBits(modifiers);
+
+            lastLineNumber = lineNumber;
+            lastStartCharacter = startCharacter;
+
+            deltaLineOut = deltaLine;
+            startCharacterDeltaOut = startCharacterDelta;
+            tokenLengthOut = tokenLength;
+            tokenTypeOut = tokenTypeIndex;
+            tokenModifiersOut = (int)modifierBits;
+        }
+
+        private static int GetTokenTypeIndex(ClassifiedSpan tokenTypeClassifiedSpan, Dictionary<string, int> tokenTypesToIndex)
+        {
+            s_classificationTypeToSemanticTokenTypeMap.TryGetValue(tokenTypeClassifiedSpan.ClassificationType, out var tokenTypeStr);
+            Debug.Assert(tokenTypeStr != null);
+
+            return tokenTypesToIndex[tokenTypeStr];
+        }
+
+        private static TokenModifiers GetTokenTypeModifierBits(IEnumerable<ClassifiedSpan> modifiers)
+        {
+            // Roslyn currently only has one modifier, but this logic should be updated if more are added
+            // in the future.
+            if (modifiers.IsEmpty())
+            {
+                return TokenModifiers.None;
+            }
+
+            Debug.Assert(modifiers.Single().ClassificationType == ClassificationTypeNames.StaticSymbol);
+            return TokenModifiers.Static;
```

Comment:
we literally only have this case.  i would just check for this against the original classified span info.